mfs.go

```diff
@@ -0,0 +1,370 @@
+		return nil
+	}
+}
+
+// Count maximum number of bytes to read
+func (filesRead) Count(count int64) filesOpt {
+	return func(rb *RequestBuilder) error {
+		rb.Option("count", count)
+		return nil
+	}
+}
+
+// Format print statistics in given format. Allowed tokens: <hash> <size> <cumulsize> <type> <childs>. Conflicts with other format options.
+func (filesStat) Format(format string) filesOpt {
+	return func(rb *RequestBuilder) error {
+		rb.Option("format", format)
+		return nil
+	}
+}
+
+// Hash print only hash. Implies '--format=<hash>'. Conflicts with other format options.
+func (filesStat) Hash(hash bool) filesOpt {
+	return func(rb *RequestBuilder) error {
+		rb.Option("hash", hash)
+		return nil
+	}
+}
+
+// Size print only size. Implies '--format=<cumulsize>'. Conflicts with other format options.
+func (filesStat) Size(size bool) filesOpt {
+	return func(rb *RequestBuilder) error {
+		rb.Option("size", size)
+		return nil
+	}
+}
+
+// WithLocal compute the amount of the dag that is local, and if possible the total size.
+func (filesStat) WithLocal(withLocal bool) filesOpt {
+	return func(rb *RequestBuilder) error {
+		rb.Option("with-local", withLocal)
+		return nil
+	}
+}
+
+// Offset byte offset to begin writing at
+func (filesWrite) Offset(offset int64) filesOpt {
+	return func(rb *RequestBuilder) error {
+		rb.Option("offset", offset)
+		return nil
+	}
+}
+
+// Create create the file if it does not exist
+func (filesWrite) Create(create bool) filesOpt {
+	return func(rb *RequestBuilder) error {
+		rb.Option("create", create)
+		return nil
+	}
+}
+
+// Parents make parent directories as needed
+func (filesWrite) Parents(parents bool) filesOpt {
+	return func(rb *RequestBuilder) error {
+		rb.Option("parents", parents)
+		return nil
+	}
+}
+
+// Truncate truncate the file to size zero before writing
+func (filesWrite) Truncate(truncate bool) filesOpt {
+	return func(rb *RequestBuilder) error {
+		rb.Option("truncate", truncate)
+		return nil
+	}
+}
+
+// Count maximum number of bytes to write
+func (filesWrite) Count(count int64) filesOpt {
+	return func(rb *RequestBuilder) error {
+		rb.Option("count", count)
+		return nil
+	}
+}
+
+// RawLeaves use raw blocks for newly created leaf nodes. (experimental)
+func (filesWrite) RawLeaves(rawLeaves bool) filesOpt {
+	return func(rb *RequestBuilder) error {
+		rb.Option("raw-leaves", rawLeaves)
+		return nil
+	}
+}
+
+// CidVersion cid version to use. (experimental)
+func (filesWrite) CidVersion(version int) filesOpt {
+	return func(rb *RequestBuilder) error {
+		rb.Option("cid-version", version)
+		return nil
+	}
+}
+
+// Hash hash function to use. Will set Cid version to 1 if used. (experimental)
+func (filesWrite) Hash(hash string) filesOpt {
+	return func(rb *RequestBuilder) error {
+		rb.Option("hash", hash)
+		return nil
+	}
+}
+
+// FilesChcid change the cid version or hash function of the root node of a given path
+func (s *Shell) FilesChcid(path string, options ...filesOpt) error {
```

Comment:
Let's take a context in all of these functions. We're trying to do that for all newly defined functions in this library.